app/src/main/java/com/example/ithaca_transit_android_v2/MainActivity.kt

```diff
@@ -1,50 +1,57 @@
 package com.example.ithaca_transit_android_v2
 
+import android.content.Context
+import android.content.pm.PackageManager
+import android.location.LocationManager
 import android.os.Bundle
-import android.util.Log
-import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
 import com.example.ithaca_transit_android_v2.models.Location
+import com.example.ithaca_transit_android_v2.presenters.MapPresenter
 import com.example.ithaca_transit_android_v2.presenters.SearchPresenter
 import com.example.ithaca_transit_android_v2.ui_adapters.SearchViewAdapter
+
+import com.example.ithaca_transit_android_v2.util.CurrLocationManager
+
 import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
 import io.reactivex.disposables.Disposable
 import kotlinx.android.synthetic.main.activity_main.*
-import kotlinx.android.synthetic.main.activity_toolbar_search.*
+
+import kotlinx.android.synthetic.main.search_main.*
+import kotlinx.android.synthetic.main.search_secondary.*
+
 
 class MainActivity : AppCompatActivity(), OnMapReadyCallback {
     private lateinit var searchDisposable: Disposable
     private var mSearchLocations: List<Location> = ArrayList()
     private lateinit var mSearchAdapter: SearchViewAdapter
     private lateinit var mSearchPresenter: SearchPresenter
+    private lateinit var mCurrLocationManager: CurrLocationManager
 
     override fun onMapReady(map: GoogleMap?) {
-        map!!.setOnMapClickListener { point ->
-            Log.i("qwerty", "map clicked")
-            search_view.clearFocus()
-        }
+        val p = MapPresenter()
+        p.initMapView(map!!)
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         mSearchAdapter = SearchViewAdapter(this, mSearchLocations)
-        mSearchPresenter = SearchPresenter(search_view, this, mSearchAdapter)
+        mSearchPresenter = SearchPresenter(search_card_holder, this, mSearchAdapter)
 
         searchDisposable = mSearchPresenter.initSearchView()
         (map_fragment as SupportMapFragment).getMapAsync(this)
 
         // set up search adapter
         locations_list.adapter = mSearchAdapter
-        locations_list.setOnItemClickListener { parent, view, position, id ->
-            val destination = parent.getItemAtPosition(position) as Location
-            // TODO: change map state to reflect destination
+        locations_list_2.adapter = mSearchAdapter
```

Comment:
why is there two locations lists lol

Reply:
Added comment. One represents the list for the initial searchview the other is the list of locations when the user is editing their route.